### ISSUE TITLE
ts-reset for DOM lib

### ISSUE THREAD

**Issue by lifeiscontent** [total-typescript/ts-reset#64]
Hey @mattpocock, thanks for ts-reset. I've been thinking about making something similar for DOM specs. Several APIs say they exist, but they do not work across all browsers.

My plan on how to solve this is to use something like Playwright, inspect all DOM APIs and check that they exist and write a JSON object out towards the end of the test runner and then make a diff and where the APIs don't work across browser make the API optional, and append some docs with notes around which browsers support what.

What do you think about this being a part of or an extension of ts-reset?

**Comment by bcheidemann** [comment]
This would really help avoid footguns in certain cases and (as an outside observer) feels like it's very much in the spirit of this project.

The only concern I have is that it's a much more empirical approach than what's already done here. It could easily be its own project.

However, my feeling is that it's much better for the ecosystem (especially new devs) to have one simple answer to this problem rather than dividing it across multiple projects. Since the project is modular, people don't need to use it if they don't want to.

Regardless if this ends up being a separate project or integrated to this one, I'd be happy to help if you want a partner in crime @lifeiscontent ✌️

TLDR; I would love to see this become a part of ts-reset and an happy to help!

**Comment by mattpocock** [comment]
@lifeiscontent I agree that there are some DOM API's that could be typed better. It's definitely in scope for ts-reset to work on those. Happy to accept issues and PR's in that vein, for individual improvements.

However, I don't want to go down the road of typing API's that TS hasn't managed to yet. I trust TS's process - it sounds similar to yours, except that it uses API data from [caniuse.com](https://caniuse.com/) to only type things which have support in 2+ browsers.

**Comment by lifeiscontent** [comment]
@mattpocock i wasn't planning on typing anything that the TS team hasn't already typed for us, but more like narrowing those types to showcase APIs that may not work across browsers (that currently exist in the DOM types)

Does that align with you?

**Comment by mattpocock** [comment]
@lifeiscontent Could you give me an example?

**Comment by lifeiscontent** [comment]
@mattpocock yes, for example all aria* attributes on an element do not work in Firefox.

**Comment by mattpocock** [comment]
I think I'd prefer not to get into that particular territory - sounds like a maintenance nightmare to keep all that stuff up-to-date, and I think TS does a good enough job.

**Comment by bcheidemann** [comment]
@lifeiscontent are you interested in making this a separate project?
